rpc/server.go

```diff
@@ -200,7 +189,7 @@ func (s *Server) serve(listener net.Listener, handler http.Handler) error {
 	s.Logger.Info("serving HTTP", "listen address", listener.Addr())
 	s.server = http.Server{
 		Handler:           handler,
-		ReadHeaderTimeout: ReadHeaderTimeout,
+		ReadHeaderTimeout: readHeaderTimeout,
 	}
```

Comment:
do we want readtimeout too? it's often recommended

Reply:
no need IMO, it quite overlaps with the timeouthandler

Reply:
If I'm not mistaken most rpc methods don't have timeout defined (just the subscribe).

however I guess we could assume that rpc nodes exposed to the public have timeouts defined in the infra level (e.g on the load balancer/reverse-proxy) etc. 

So practically speaking not sure it's a concern. with that being said there is also no arm in adding it I guess.